### PR TITLE
Modal Components

### PR DESCRIPTION
**Pull request by cenwachukwu**
1. I added a this.prop.children for adding the name of the group user wants to join or the person they want to follow.

### CODE REVIEW

**Comment by ZakkMan** [comment]
@cenwachukwu make sure you assign yourself the PR

File: client/src/components/Modal/Modal.css [hunk 1]
```diff
@@ -0,0 +1,73 @@
+@import url('https://fonts.googleapis.com/css?family=Reem+Kufi&display=swap');
```
Comment (ZakkMan) [inline client/src/components/Modal/Modal.css]: This should be set globally

File: client/src/components/Modal/Modal.css [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    width: 582px;
+    height: 309.39px;
+    background: #F2BC0F;
+    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
+    border-radius: 3px;
+    display: flex;
+    flex-direction: column;
+}
+
+.close-modal-btn {
```
Comment (ZakkMan) [inline client/src/components/Modal/Modal.css]: Can you use the full name (i.e. `button` rather than `btn`)

File: client/src/components/Modal/Modal.css [hunk 3]
```diff
@@ -0,0 +1,73 @@
+@import url('https://fonts.googleapis.com/css?family=Reem+Kufi&display=swap');
+
+.Modal {
+    margin: 0;
+    padding: 0;
+    box-sizing: border-box;
+    width: 582px;
+    height: 309.39px;
```
Comment (ZakkMan) [inline client/src/components/Modal/Modal.css]: Rather than set an explicit height and width, can you use percentages and set a the padding?

File: client/src/components/Modal/Modal.css [hunk 4]
```diff
@@ -0,0 +1,73 @@
+@import url('https://fonts.googleapis.com/css?family=Reem+Kufi&display=swap');
+
+.Modal {
+    margin: 0;
+    padding: 0;
+    box-sizing: border-box;
```
Comment (ZakkMan) [inline client/src/components/Modal/Modal.css]: This should be set globally

File: client/src/components/Modal/Modal.css [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    width: 582px;
+    height: 309.39px;
+    background: #F2BC0F;
```
Comment (ZakkMan) [inline client/src/components/Modal/Modal.css]: Can you use teh CSS variables for colors and font properties?

File: client/src/components/Modal/Modal.css [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    float: right;
+    font-size: 65px;
+    margin-top: -20px;
+}
+
+.modal-body {
+    margin: 0 auto;
+}
+
+.htag{
```
Comment (ZakkMan) [inline client/src/components/Modal/Modal.css]: What does this style? Can you give this a semantic class name?

File: client/src/components/Modal/Modal.css [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+.modal-footer{
+    margin-top: 75px;
+    height: 67.51px;
+    display: flex;
+    flex-direction: row;
+}
+
+.confirm{
+    border: 1px solid rgba(250, 250, 250, 0.25);
+    width: 285.09px;
```
Comment (ZakkMan) [inline client/src/components/Modal/Modal.css]: There are no fractions on pixels - only absolute numbers

File: client/src/components/Modal/Modal.js [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+            <button className="cancel">Cancel</button>
+          </div>
+          <div className="confirm">
+            <button className="confirm">Confirm</button>
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
```
Comment (ZakkMan) [inline client/src/components/Modal/Modal.js]: You can shorten this up a lot - this is a lot of duplication. The text of the button can come in as a prop and then. Remember also that you want to accept props for what happens when someone clicks on cancel, the X, or confirm!

File: client/src/components/Modal/Modal.stories.js [hunk 9]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import Modal from './Modal';
+
+storiesOf('Modal', module)
+  .add("Follow Modal", () => 
+    <Modal/>
```
Comment (ZakkMan) [inline client/src/components/Modal/Modal.stories.js]: We want to render something like this:

```jsx
<Modal message={"Are you sure you want to Join?"} confirmButtonText={"Join"} onConfirm={this.handleConfirm} />
```

File: client/src/components/Modal/Modal.css (hunk 4 above)
Comment (ZakkMan) [reply client/src/components/Modal/Modal.css]: @cenwachukwu did you fix this one?

File: client/src/components/Modal/Modal.css [hunk 10]
```diff
@@ -0,0 +1,83 @@
+@import url('https://fonts.googleapis.com/css?family=Roboto&display=swap');
```
Comment (ZakkMan) [inline client/src/components/Modal/Modal.css]: The font is provided globally, so you don't need to import it - just use the CSS variables

File: client/src/components/Modal/Modal.css [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+
+*{
+    margin: 0;
+    padding: 0;
+    box-sizing: border-box;
+    font-family: var(--default-font);
+}
```
Comment (ZakkMan) [inline client/src/components/Modal/Modal.css]: Can you delete this? It'll be in `App.css`

File: client/src/components/Modal/Modal.css [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+    width: 582px;
+    height: 309px;
+    background: var(--yellow);
+    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
+    border-radius: 3px;
+    display: flex;
+    flex-direction: column;
+}
+
+.close-modal-button {
```
Comment (ZakkMan) [inline client/src/components/Modal/Modal.css]: Can you update your class names to be in line with the BEM naming convention?

File: client/src/components/Modal/Modal.css [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+    color: black;
+    cursor: pointer;
+    float: right;
+    font-family: 'Roboto', sans-serif;
```
Comment (ZakkMan) [inline client/src/components/Modal/Modal.css]: Can you use the variables here for all font properties and color properties?

File: client/src/components/Modal/Modal.css [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
+    border-radius: 3px;
+    display: flex;
+    flex-direction: column;
+}
+
+.close-modal-button {
+    color: black;
+    cursor: pointer;
+    float: right;
```
Comment (ZakkMan) [inline client/src/components/Modal/Modal.css]: Can you use flexbox instead of floats?

File: client/src/components/Modal/Modal.css [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+    background: var(--yellow);
+    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
+    border-radius: 3px;
+    display: flex;
+    flex-direction: column;
+}
+
+.close-modal-button {
+    color: black;
+    cursor: pointer;
```
Comment (ZakkMan) [inline client/src/components/Modal/Modal.css]: Can you change this to an `a` tag so you don't have to set the cursor manually

File: client/src/components/Modal/Modal.css [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+    height: 60%;
+}
+
+.modal-message{
+    width: 90%;
+    height: 100%;
+    margin: 0 auto;
+}
+
+h3{
```
Comment (ZakkMan) [inline client/src/components/Modal/Modal.css]: Please don't set styles on tags inside a component - use BEM

File: client/src/components/Modal/Modal.css [hunk 17]
```diff
@@ -0,0 +1,83 @@
...
+    border: 1px solid rgba(37, 37, 37, 0.25);
+    width: 100%;
+    height: 100%;
+    font-size: 24px;
+    line-height: 36px;
+    text-align: center;
+    color: #000000;
+    text-transform: uppercase;
+    background: rgba(37, 37, 37, 0.25);
+}
```
Comment (ZakkMan) [inline client/src/components/Modal/Modal.css]: Are you able to use the `Button` component?

File: client/src/components/Modal/Modal.js [hunk 18]
```diff
@@ -0,0 +1,43 @@
+import React, { Component } from 'react';
+import "./Modal.css";
+import "../../App.css";
```
Comment (ZakkMan) [inline client/src/components/Modal/Modal.js]: Can you delete this?

File: client/src/components/Modal/Modal.js [hunk 19]
```diff
@@ -0,0 +1,43 @@
...
+
+
+// Class Based React Component
+class Modal extends Component {
+  constructor(props) {
+    super(props);
+    console.log(props);
```
Comment (ZakkMan) [inline client/src/components/Modal/Modal.js]: Can you remove any `console.log` statements?

File: client/src/components/Modal/Modal.js [hunk 20]
```diff
@@ -0,0 +1,43 @@
...
+
+    // Default CSS class to apply to the Component
+    this.state = {
+      classList: "Modal",
+    };
+  }
```
Comment (ZakkMan) [inline client/src/components/Modal/Modal.js]: Please just put the class name in the `className` attribute

File: client/src/components/Modal/Modal.js [hunk 21]
```diff
@@ -0,0 +1,43 @@
+import React, { Component } from 'react';
+import "./Modal.css";
+import "../../App.css";
+
+
+// Class Based React Component
+class Modal extends Component {
```
Comment (ZakkMan) [inline client/src/components/Modal/Modal.js]: I think this could a function component

File: client/src/components/Modal/Modal.js [hunk 22]
```diff
@@ -0,0 +1,43 @@
...
+    super(props);
+    console.log(props);
+
+    // Default CSS class to apply to the Component
+    this.state = {
+      classList: "Modal",
+    };
+  }
+  // Runs after Component is loaded in the broswer
+  componentDidMount() { }
```
Comment (ZakkMan) [inline client/src/components/Modal/Modal.js]: Can you delete this, since you're not using it

File: client/src/components/Modal/Modal.stories.js [hunk 23]
```diff
@@ -0,0 +1,39 @@
...
+  alert('Modal Closed!')
+}
+
+storiesOf('Modal', module)
+  .add("Follow Modal", () => 
+    <Modal message={"Are you sure you want to follow"} 
+          confirmText={"Confirm"} 
+          cancelText={"Cancel"}
+          onConfirm={handleConfirm} 
+          onCancel={handleCancel}
```
Comment (ZakkMan) [inline client/src/components/Modal/Modal.stories.js]: I think `onCancel` and `onClose` are the same - we should invoke the same callback when they click Cancel as when they click the X

File: client/src/components/Modal/Modal.css (hunk 12 above)
Comment (ZakkMan) [inline client/src/components/Modal/Modal.css]: I think this needs a little more padding around it

Screen Shot 2019-12-11 at 10 41 15 AM

File: client/src/components/Modal/Modal.stories.js [hunk 24]
```diff
@@ -0,0 +1,39 @@
...
+function handleCancel (evt){
+  evt.preventDefault()
+  alert('Canceled!')
+}
+function handleClose (evt){
+  evt.preventDefault()
+  alert('Modal Closed!')
+}
+
+storiesOf('Modal', module)
```
Comment (ZakkMan) [inline client/src/components/Modal/Modal.stories.js]: Can you work through and create a story for having the modal pop up and close out? Like just a button or link that we can click that triggers the open and then with close functions

**Comment by caioingber** [comment]
A bit confused on what still needs to be worked on for this PR. Any ideas?

**Comment by caioingber** [comment]
A bit confused on what still needs to be worked on for this PR - seems like it is functional and responsive in storybook. Any ideas?

File: client/src/components/FunctionalModal/FunctionalModal.css [hunk 25]
```diff
@@ -0,0 +1,29 @@
+.FunctionalModal{
```
Comment (ZakkMan) [inline client/src/components/FunctionalModal/FunctionalModal.css]: Can you make the class names consistent with the CSS naming convention?

File: client/src/components/Modal/Modal.stories.js (hunk 24 above)
Comment (caioingber) [reply client/src/components/Modal/Modal.stories.js]: Hey @ZakkMan from what I can tell, the `FunctionalModal` component is the one that demonstrates functionality as it includes a button and the Modal component itself, and it is where state lives so that showModal prop can be toggle true/false

**Comment by caioingber** [comment]
modal was changed to a functional component, updated styling names to match convention. See comment above in regards to FunctionalModal vs. Modal

**Comment by caioingber** [comment]
Set up a ModalContainer.js file in the Modal directory - per our convo, this contains both the functional Modal component as well as a Button component which allows the user to open the modal. Updated stories to utilize the ModalContainer component to demonstrate functionality. If I understood correctly, this is what you were getting at. Leaving the FunctionalModal component for now, in case there are issues with my updates.

File: client/src/components/Modal/Modal.css [hunk 26]
```diff
@@ -0,0 +1,125 @@
...
+  text-decoration: none;
+  color: black;
+}
+
+.Modal__body {
+  width: 100%;
+  height: 60%;
+}
+
+.Modal__messageBox {
```
Comment (ZakkMan) [inline client/src/components/Modal/Modal.css]: With the naming convention we're using, I think this should be `.Modal__message-box`

Comment (ZakkMan) [reply client/src/components/Modal/Modal.css]: (Sorry for not catching this earlier 😳 )

File: client/src/components/Modal/Modal.css [hunk 27]
```diff
@@ -0,0 +1,125 @@
...
+  height: 23%;
+  display: flex;
+  flex-direction: row;
+}
+
+.Modal__footerButtons {
+  width: 100%;
+}
+
+.Modal__footerButtonConfirm {
```
Comment (ZakkMan) [inline client/src/components/Modal/Modal.css]: Does the button component not work here? It's okay if it doesn't but that would be ideal

File: client/src/components/Modal/Modal.js [hunk 28]
```diff
@@ -0,0 +1,39 @@
...
+
+const Modal = props => {
+  return (
+    <div
+      className="Modal"
+      style={{
+        transform: props.showModal ? "translateY(0vh)" : "translateY(-100vh)",
+        opacity: props.showModal ? "1" : "0"
+      }}
+    >
```
Comment (ZakkMan) [inline client/src/components/Modal/Modal.js]: This is a good way of handling this. Another is to create 'state' classes like `.is-visible` or `.is-hidden` and then toggle adding or removing those classes.

Comment (ZakkMan) [reply client/src/components/Modal/Modal.js]: You don't have to go with that

File: client/src/components/Modal/ModalContainer.js [hunk 29]
```diff
@@ -0,0 +1,54 @@
+import React, { Component } from "react";
```
Comment (ZakkMan) [inline client/src/components/Modal/ModalContainer.js]: Do you mind putting an underscore in the beginning of the file name here? I forgot to mention that when we spoke on slack. That's a way of signalling to other developers that this is a "hidden" file/class/whatever. That will make it extra clear that people shouldn't use this component

Comment (ZakkMan) [reply client/src/components/Modal/ModalContainer.js]: Just to the file name

File: client/src/components/Modal/ModalContainer.js [hunk 30]
```diff
@@ -0,0 +1,54 @@
...
+  };
+
+  handleOpenModal = () => {
+    this.setState({
+      modalOpen: true
+    });
+  };
+
+  render() {
+    console.log(this.state.modalOpen);
```
Comment (ZakkMan) [inline client/src/components/Modal/ModalContainer.js]: Can you remove this `console.log` - we never want to push that to prod

File: client/src/components/Navbar/Navbar.js [hunk 31]
```diff
@@ -15,13 +15,13 @@ class Navbar extends Component {
     this.state = {
```
Comment (ZakkMan) [inline client/src/components/Navbar/Navbar.js]: Do we need these changes? They seem unrelated to what you're working on? Can you maybe reset this file so these changes aren't part of this PR?

File: client/src/index.js [hunk 32]
```diff
@@ -5,6 +5,7 @@ import App from "./App";
 import * as serviceWorker from "./serviceWorker";
 import { BrowserRouter as Router } from "react-router-dom";
 
+
```
Comment (ZakkMan) [inline client/src/index.js]: Same with this change - can you reset it so its not in this PR?

File: client/src/components/Modal/Modal.css (hunk 27 above)
Comment (caioingber) [reply client/src/components/Modal/Modal.css]: From what I can tell, the styling for these buttons are quite specific (different color, borders, sizes) from what is prescribed in the storybook buttons, and they appear to be standalone

File: client/src/components/Modal/Modal.js (hunk 28 above)
Comment (caioingber) [reply client/src/components/Modal/Modal.js]: Yea, whoever in 33 was working on it, implemented that functionality, it's kinda nice since it allows for a nice sliding transition

**Comment by caioingber** [comment]
This should be fixed, the changes from app, index and navbar all came from merging develop locally, so there shouldn't be any changes to the develop branch. Confirmed storybook path is correct Layout/Modal. Resolved previous conversations about naming conventions, etc
